### PR TITLE
Fix #1733. Force to obtain new tokens if accessToken is nil

### LINKED ISSUE #1733: [AWSCognitoAuth]  idToken and accessToken from session are nil after user attributes update
**Describe the bug**

We are using Cognito and AWSMobileClient to authenticate users in one of our apps. This is a code snippet showing how we are authenticating the user: 

After sign in the user, we perform an update of its attributes in order to update its nickname. After this step, `idToken` and `accessToken` from `AWSMobileClient.getTokens()` method are nil.

I have debugged the SDK and the problem is caused by `updateAttributes` method of `AWSCognitoIdentityUser` class. When this method is executed, the `clearSession` method is called, causing `accessToken` and `idToken` to be nil. 

The problem happens when you try to retrieve the new tokens. The method used to achieve it is `getSessionInternal` of `AWSCognitoAuth` class. In this method, the access token and id token are only updated if the expiration date has arrived. This cause the tokens to remain nil after and updateAttributes

**To Reproduce**
1.- Create a sample app.
2.- Include these dependencies in your podfile
  ```
pod 'AWSAPIGateway'
pod 'AWSMobileClient'
pod 'AWSAuthUI'
pod 'AWSUserPoolsSignIn'
pod 'AWSAppSync'
pod 'AWSPinpoint'
```

3.- In your app delegate initialize the AWSMobileClient
```
 AWSMobileClient.sharedInstance().initialize { (userState, error) in
            
            if let userState = userState {
                print("UserState: \(userState)")
            }
            if let error = error {
                print("Error: \(error)")
            }
 }
```

4.- Sign in using Hosted UI
```
let hostedUIOptions : HostedUIOptions? = HostedUIOptions(idpIdentifier: "idpidentifier ")
AWSMobileClient.sharedInstance().showSignIn(navigationController: self.navigationController!, hostedUIOptions: hostedUIOptions) { (userState, error) in
            if let error2 = error as? AWSMobileClientError {
                print(error2.localizedDescription)
            }
            if let userState = userState {
                print("Status: \(userState.rawValue)")
            }
  }
```

5.- [THIS IS THE KEY STEP] Update user attributes:
```
 let pool = AWSCognitoIdentityUserPool.default()
        let nicknameAttribute = AWSCognitoIdentityUserAttributeType(name: "nickname", value: newNickname)

        pool.currentUser()?.update([nicknameAttribute]).continueWith(block: { (task) -> Any? in
            DispatchQueue.main.async {
                if task.error == nil {
                    onResult(.success(()))
                } else {
                    onResult(.failure(MigoError(type: UserError.Unknown)))
                }
            }
        })
```

6.- Try to get tokens
```
   AWSMobileClient.sharedInstance().getTokens { (tokens, error) in
    // tokens.idToken? is nil
   // tokens.accessToken is nil
}
```

**Which AWS service(s) are affected?**

AWSCognitoAuth, AWSMobileClient

**Expected behavior**

I expect  `idToken`and `accessToken` to have value

**Environment(please complete the following information):**
 - SDK Version: 2.10.2
 - Dependency Manager: Cocoapods
 - Swift Version : 5.0

**Device Information (please complete the following information):**
 - Device: iPhone XS Simulator
 - iOS Version: 12.2

### PR DESCRIPTION
**Pull request by JesusMartinAlonso**
Modifications in `GetSessionInternal` method to force to obtain new tokens if accessToken is nil

*Issue #, if available:* #1733 

*Description of changes:*

Retrieve access token value in order to check if it is null. In that case, we have to retrieve new tokens. This situation can happen after calling `clearSession` method.

See #1733 for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by palpatim** [comment]
This will need a test before merging

**Comment by palpatim** [comment]
We'll need to understand the intent behind the original behavior. I don't immediately see why we'd call `clearSession` during a user attribute refresh, but it does make sense that a `clearSession` invocation would require us to revalidate tokens.

**Comment by JesusMartinAlonso** [comment]
@palpatim Is there an estimated time? Note that is a serious bug: Our app won't work properly after this will solved.

Don't hesitate to ask me if you have any doubts.

Thank you!

**Comment by JesusMartinAlonso** [comment]
Any updates about this PR?

**Comment by javiboo** [comment]
@palpatim When do you plan to release the fix of this issue?

File: AWSCognitoAuth/AWSCognitoAuth.m [hunk 1]
```diff
@@ -336,9 +336,11 @@ - (void)getSessionInternal: (nullable UIViewController *) vc completion: (AWSCog
...
             NSString * refreshToken = [self refreshTokenFromKeyChain:keyChainNamespace];
+            NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoAuthUserAccessToken];
+            NSString * accessToken = self.keychain[accessTokenKey];
 
             //if the session expires > 5 minutes return it.
-            if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending){
+            if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending && accessToken){
                 NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoAuthUserIdToken];
                 NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoAuthUserAccessToken];
                 AWSCognitoAuthUserSession * session = [[AWSCognitoAuthUserSession alloc] initWithIdToken:self.keychain[idTokenKey] accessToken:self.keychain[accessTokenKey] refreshToken:refreshToken expirationTime:expiration];
```
Comment (palpatim) [inline AWSCognitoAuth/AWSCognitoAuth.m]: Minor code duplication: can replace with the variables declared on 339 & 340.

Comment (JesusMartinAlonso) [reply AWSCognitoAuth/AWSCognitoAuth.m]: I don't understand really well. What change do you suggest?

Comment (royjit) [reply AWSCognitoAuth/AWSCognitoAuth.m]: Created a PR for the test case - https://github.com/aws-amplify/aws-sdk-ios/pull/1940.

**Pull request merged**